core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -916,13 +958,13 @@ private[spark] object JsonProtocolSuite extends Assertions {
   }
 
   private def assertJsonStringEquals(expected: String, actual: String, metadata: String): Unit = {
-    val expectedJson = parse(expected)
-    val actualJson = parse(actual)
+    val expectedJson = mapper.readTree(expected)
+    val actualJson = mapper.readTree(actual)
     if (expectedJson != actualJson) {
       // scalastyle:off
       // This prints something useful if the JSON strings don't match
-      println(s"=== EXPECTED ===\n${pretty(expectedJson)}\n")
```

Comment:
I should probably find a way to preserve the pretty formatting in the debug output. Maybe I can configure the mapper used here in the tests?